lib/Net/Async/Github.pm

```diff
@@ -247,6 +247,83 @@ sub Net::Async::Github::Repository::grant_team {
     )
 }
 
+=head2 create_branch
+
+Creates a new branch.
+
+Takes the following named parameters:
+
+=over 4
+
+=item * C<owner> - which organisation owns the target repository
+
+=item * C<repo> - the repository to raise the PR against
+
+=item * C<branch> - new branch name that will be created
+
+=item * C<sha> - the SHA1 value for this branch
+
+=back
+
+=cut
+
+sub create_branch {
+    my ($self, %args) = @_;
+    $self->validate_args(%args);
+    $self->http_post(
+        uri => $self->endpoint(
+            'git_refs_create',
+            owner => $args{owner},
+            repo  => $args{repo},
+        ),
+        data => {
+            ref => "refs/heads/$args{branch}",
+            sha => $args{sha}
+        },
+    )
+}
+
+
+=head2 update_ref
+
+Update a reference to a new commit
+
+Takes the following named parameters:
+
+=over 4
+
+=item * C<owner> - which organisation owns the target repository
+
+=item * C<repo> - the repository to raise the PR against
+
+=item * C<ref> - ref name that we are updating.
+
+=item * C<sha> - the SHA1 value of comment that the ref will point to
+
+=item * C<force> - force update ref even if it is not fast-forward if it is true.
+
+=back
+
+=cut
+
+sub update_ref {
+    my ($self, %args) = @_;
+    $self->validate_args(%args);
+    $self->http_patch(
+        uri => $self->endpoint(
+            'git_refs',
+            owner => $args{owner},
+            repo  => $args{repo},
+            category => 'heads',
```

Comment:
what do you mean by `category => 'heads' `?

Reply:
It's Net::Async::Github's format.  The url of this request is like this: 
`https://api.github.com/repos/octocat/Hello-World/git/refs/heads/feature-a`
Here the `heads` is the category.  `tags` is another possible category.

https://docs.github.com/en/free-pro-team@latest/rest/reference/git#list-matching-references